torch/nn/modules/activation.py

```diff
@@ -756,30 +767,25 @@ def forward(self, query, key, value, key_padding_mask=None,
         - attn_output_weights: :math:`(N, L, S)` where N is the batch size,
           L is the target sequence length, S is the source sequence length.
         """
-        if hasattr(self, '_qkv_same_embed_dim') and self._qkv_same_embed_dim is False:
+        if not self._qkv_same_embed_dim:
```

Comment:
The above check is to avoid backward compatibility problem. There may be some models trained by the old version of nn.MHA, which has no _qkv_same_embed_dim attribute.
 However, since we have released pytorch 1.2, can we finally retire the support for this? @cpuhrsch

Reply:
Does this block JIT-ing? If so, why?

Reply:
The `hasattr` call does, and really that shouldn't be happening in `forward` anyways, that's something that can be determined in `_load_state_dict()` like was suggested [here](https://github.com/pytorch/pytorch/pull/21288#discussion_r289941876)

Reply:
Since we add the warning message for the BC breaking in PyTorch 1.2 release, IMO, we could drop the support now. How's do you feel about it? @cpuhrsch

Reply:
@zhangguanheng66  - let's move this mechanism into _load_state_dict like was originally suggested and then keep it around for now. The BC breaking change should we a separate PR so we can easily keep track of it.

Reply:
In the old model, there is no '_qkv_same_embed_dim' attribute. However, I cannot use `_load_state_dict()` func to add `_qkv_same_embed_dim`. The reason is that `_qkv_same_embed_dim` is not in the state_dict of the new model. Is there a way that I can add a new attribute to an old model?

Reply:
never mind. I figure out a way to walk around. After some tests, I will submit a PR first.

Reply:
@driazati I chat this with Soumith and we feel that the fix in #24404 may actually confuse our users. 

If the PR is blocked only by the `hasattr` func, could you add it to your JIT codebase? We saw `hasattr` func are applied many places (like `quantization`). It will be required anyway in the future.